sympy/combinatorics/homomorphisms.py

```diff
@@ -176,10 +215,16 @@ def homomorphism(domain, codomain, gens, images=[]):
     If the given images of the generators do not define a homomorphism,
     an exception is raised.
 
+    If `check` is `False`, don't check whether the given images actually
+    define a homomorphism.
+
     '''
-    if isinstance(domain, PermutationGroup):
-        raise NotImplementedError("Homomorphisms from permutation groups are not currently implemented")
-    elif not isinstance(domain, (FpGroup, FreeGroup)):
+    if check and isinstance(domain, PermutationGroup):
+        raise NotImplementedError("Checking if the homomorphism is well-defined\
```

Comment:
It is better to build the string by concatenation. Otherwise all the whitespace in the beginning of lines will be included.
```
    NotImplementedError("Checking if the homomorphism is well-defined "  # note the space at end
               "<next line with ending space>"
               "<last line>")